manager/controllers/app/data_catalog_interface.go

```diff
@@ -30,18 +32,23 @@ func (r *FybrikApplicationReconciler) RegisterAsset(assetID string, catalogID st
 	}
 
 	var resourceMetadata datacatalog.ResourceMetadata
-	if info.ResourceMetadata != nil {
-		resourceMetadata = *info.ResourceMetadata.DeepCopy()
+	if resourceMetadataFromRequirements != nil {
+		resourceMetadata = *resourceMetadataFromRequirements.DeepCopy()
```

Comment:
DeepCopy is applied twice for the same object

Reply:
will fix that thanks